forge-gui/res/cardsfolder/a/aethersquall_ancient.txt

```diff
@@ -6,4 +6,6 @@ K:Flying
 T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigEnergy | TriggerDescription$ At the beginning of your upkeep, you get {E}{E}{E} (three energy counters).
 SVar:TrigEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ 3
 A:AB$ ChangeZoneAll | Cost$ PayEnergy<8> | ChangeType$ Creature.StrictlyOther | SorcerySpeed$ True | Origin$ Battlefield | Destination$ Hand | SpellDescription$ Return all other creatures to their owners' hands. Activate only as a sorcery.
+DeckHas:Ability$Counters
+DeckHints:Ability$Counters
```

Comment:
especially the hints seem too generic for energy imo 🤔 

I guess `Ability$Proliferate` is an alternative but honestly just because it's possible we really shouldn't flood every counter producer with that (especially in cases like this card which already works not too bad on its own)
Not sure what could be the right boundary - maybe only the really strong synergy ones like _Simic Ascendancy_?

FWIW `Ability$Energy` is also used on 3 cards so that's useless and should probably be removed instead (or is Energy enough of an Archetype to bother with completing it?)

Perhaps we can try and write down some rough decision guidance while working through these PR...

Reply:
Fair on being more judicious about it. I have been more inclined to hint proliferate on cards that distribute counters.

Reply:
While reviewing keywords with issues to address, I tagged:
- scripts with awaken with the creature type and keyword ability
- scripts with a level up cost of 3 mana or more with proliferate

As for energy, I feel something like `Ability$Counters.Energy` to go alongside `Ability$Counters.P1P1` could be more useful. Of course, some character other than a period can be used as a separator for this layer of classification.

Reply:
hmn working towards some Ability groups like that would also let you use logic where a rather generic tag can be fulfilled with more specific ones based on it 🥇

`DeckHints:Ability$Counters` ➡️ `DeckHas:Ability$Counters.Energy` + `DeckHas:Ability$Counters`
but
`DeckHints:Ability$Counters.Energy` ➡️ `DeckHas:Ability$Counters.Energy`

I already added the basics for that in #564

Reply:
Oh okay then, I'll propagate that with (I hope) the required judiciousness.

Also, while we're on Ability groups, I've had the vague notion that `Mana.Prismatic`/`Mana.Colors`/`Mana.AnyColor` for the likes of Birds of Paradise could be useful to go along with `Mana.Colorless`.

Reply:
Just to confirm: from looking over the code, when we're dealing with multiple kinds of counters (for instance, [Phyrexian Hydra](https://scryfall.com/card/mbs/85/phyrexian-hydra) ) the correct script snippet is `Ability$Counters.M1M1|Counters.Poison`, right?

Reply:
that should fit

Reply:
A fair bit. I think I've got the `M1M1` and `P1P1` counters adequately covered. Now it's charge counters, maybe time counters and a smattering of other aesthetic ones afterward. And then it's making the pass to recheck `Proliferate` for relevance.

Reply:
woah alright, you didn't have to take care of all types at once 💪